discovery/provider.go

```diff
@@ -17,64 +19,74 @@ var (
 	// endpoints for a certain repository.
 	ErrEndpointsNotFound = errors.NewKind("endpoinds not found for %s")
 
-	// ErrProviderStop is returned when a provider has been stopped.
-	ErrProviderStop = errors.NewKind("provider stopped")
-
-	// ErrNewRepositoriesNotFound is returned when there aren't new repositories
-	// in the organization.
-	ErrNewRepositoriesNotFound = errors.NewKind("couldn't find new repositories")
+	// ErrNewRepositoriesNotFound is returned when there aren't new
+	// repositories in the organization.
+	ErrNewRepositoriesNotFound = errors.NewKind(
+		"couldn't find new repositories")
 )
 
-// Provider interface represents the service to retrieve repositories from a
-// organization.
-type Provider interface {
-	Start() error
-	Stop() error
-}
-
 // GHProviderOpts represents configuration options for a GHProvider.
 type GHProviderOpts struct {
 	HTTPTimeout    time.Duration
 	ResultsPerPage int
 	TimeNewRepos   time.Duration
+	StopTimeout    time.Duration
 }
 
-// GHProvider will retrieve the information for all the repositories for the given
-// github organization.
+// GHProvider is a gitcollector.Provider implementation. It will retrieve the
+// information for all the repositories for the given github organization
+// to produce gitcollector.Jobs.
 type GHProvider struct {
```

Comment:
It looks like `GHProvider` is a generic provider implementation and `orgReposIter` is the specific github implementation. Probably now it's not the moment but we can open an issue to rename them and extract `orgReposIter` out of `GHProvider` so it can be reused for other providers.

Reply:
@jfontan issue opened at  https://github.com/src-d/gitcollector/issues/8

Reply:
I was thinking about this, and it could be a generic implementation for differents github providers using different endpoints (we list repos by orgs, but we could list public repos or whatever is in github's api), but it couldn't be a generic implementation for any kind of provider because it relies on the abstractions of the github client. For example the iterator returns a `*github.Repository`.